docs/source/tutorials/token_classification/entity_recognition_training.ipynb

```diff
@@ -0,0 +1,737 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "237ff9c2",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "True\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(modified(given_words, sentence_tokens)) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0b498649",
+   "metadata": {},
+   "source": [
+    "Given that some characters have been modified, we need to further map `sentence_tokens` back to the original characters. Note that such mapping is different for different models, and is usually not required by most tokenizers. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "a09fe4c5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# collapse this \n",
+    "replace = [('#', ''), ('``', '\"'), (\"''\", '\"')] \n",
+    "sentence_tokens = [[process_token(token, replace) for token in sentence_tokens[i]] for i in range(len(sentences))] \n",
+    "\n",
+    "for i in range(len(sentences)): \n",
+    "    short = ''.join(given_words[i]) \n",
+    "    if \"''\" in short: \n",
+    "        processed_tokens, processed_probs = [], [] \n",
+    "        for token, prob in zip(sentence_tokens[i], sentence_probs[i]): \n",
+    "            if token != '\"': \n",
+    "                processed_tokens.append(token) \n",
+    "                processed_probs.append(prob) \n",
+    "            else: \n",
+    "                for _ in range(2): \n",
+    "                    processed_tokens.append(\"'\") \n",
+    "                    processed_probs.append(prob) \n",
+    "        sentence_tokens[i] = [token for token in processed_tokens] \n",
+    "        sentence_probs[i] = np.array([prob for prob in processed_probs]) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00edbe08",
+   "metadata": {},
+   "source": [
+    "In this example, we are more interested in severe types of mislabels, such as `B-LOC` vs. `B-PER`, instead of `B-LOC` vs. `I-LOC`. Therefore, we discard the `B-` and `I-` prefixes, and get the model-predictive probabilities for each subword-level token. The merged entities are `[O, MIS, PER, ORG, LOC]`. As shown below: \n",
+    "\n",
+    "- `given_maps` is an array of length equal to the original number of entities, such that `given_maps[i]` is the mapped entity of the i'th entity \n",
+    "- `model_maps` is an array of length equal to the number of model predicted labels, such that `model_maps[i]` is the mapped entity of the i'th model predicted entity. If `model_maps[i] < 0`, it indicates that the entity does not map to a valid named entity. This usually occurs when the model predicted entities include start/end tags. If `np.any(model_maps < 0)`, `pred_probs` will be normalized internally in `merge_probs`. "
```

Comment:
Before you clarify that `pred_probs` will be normalized, you should clarify why `merge_probs` does. I would explicitly state the operations including the final one being renormalization to ensure `pred_probs` sums to 1 (over the possible classes) for each token.